src/AbstractSelfHelp.java

```diff
@@ -0,0 +1,55 @@
+import java.util.ArrayList;
+
+public abstract class AbstractSelfHelp {
+	private SelfHelpDescription selfHelpDescription;
+
+	public abstract String getSelfHelp();
+
+	public int getSeverityDecrease(SelfHelpDescription x) {
+		return 1;
+	}
+
+	public SelfHelpDescription getSelfHelpDescription() {
+		return selfHelpDescription;
+	}
+
+	public void setSelfHelpDescription(SelfHelpDescription selfHelpDescription) {
+		this.selfHelpDescription = selfHelpDescription;
+	}
+	
+	public static SelfHelpDescription getAverageEffectivity(ArrayList<AbstractSelfHelp> selfHelpList){
+		if(selfHelpList.size() == 0){
+			return SelfHelpDescription.DIDNT_HELP;
+		}
+		double numUses = selfHelpList.size();
+		double effectivitySum = 0;
+		for(AbstractSelfHelp selfHelp : selfHelpList){
+			if(selfHelp.getSelfHelpDescription() == SelfHelpDescription.HELPED_A_LOT){
+				effectivitySum += 4;

```

Comment:
The numeric values that correspond to the qualitative labels could be included in the enum definition for SelfHelpDescription by introducing a constructor in the enum that takes those values.  Note that the enum.ordinal() method can't be repurposed for this since it starts at zero.